### DISCUSSION TITLE
Access to Global User Variable in Insights

### DISCUSSION THREAD

**Discussion by HelloImZomi** [directus/directus#20676] [Feature Requests]
### Summary

I would like to have access to a global variable, perhaps named {{ USER }} or {{ MY_ACCOUNT }}, that allows me to retrieve all the data of my user account. This is intended to personalize and refine the statistics based on the user attempting to view them. Preferably, this should not only fetch the user's ID but all the data from the directus_users collection of my current session.

### Basic Example

For instance, being able to use {{ USER.id }} to filter a list where the current user created the record and is the only one able to view or edit it. Also, using {{ USER.role }} to identify if the user holds a specific role, like viewing certain values.

### Motivation

- Personalized statistics and lists.
- Easier access to data.
- Improved privacy.

### Detailed Design

What is required is that it has the same structure as the global variables in insights, but with the difference that this one is already connected to the current user's collection to obtain their data. It should not only return the current user's ID when calling {{ USER }} or {{ MY_ACCOUNT }}, but we should also be able to access all their data, for example, {{ USER.email }}, {{ USER.tags }}, {{ USER.role }}, and any other additional field that has been added.

### Must Have:
- Access data through the global variable {{ USER }} or {{ MY_ACCOUNT }}
- Retrieve data of the current user session

### Drawbacks

High complexity in development to obtain current user's data.

### Alternatives

I considered creating a global relationship variable that queries the directus_users collection and filters data based on the selected user. However, this action must be performed every time a user wants to consult their information, which also affects user privacy because they could see data of other users when the goal is to restrict access.

### Adoption Strategy

This change does not affect the system's database structure, but rather it is in the logical part where it is required to enable obtaining the current user's data.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
